proto/regen/ecocredit/v1alpha1/genesis.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+package regen.ecocredit.v1alpha1;
+
+import "regen/ecocredit/v1alpha1/types.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/regen-network/regen-ledger/x/ecocredit";
+
+// GenesisState defines ecocredit module's genesis state.
+message GenesisState {
+  // Params contains the updateable global parameters for use with the x/params
+  // module
+  Params params = 1 [ (gogoproto.nullable) = false ];
+
+  // class_info is the list of credit class info.
+  repeated ClassInfo class_info = 2;
+
+  // batch_info is the list of credit batch info.
+  repeated BatchInfo batch_info = 3;
+
+  // id_seq is used to get next class/batch id.
+  uint64 id_seq = 4;
+
+  // tradable_balances is the list of credit batch tradable units.
+  repeated Balance tradable_balances = 5;
+
+  // retired_balances is the list of credit batch retired units.
+  repeated Balance retired_balances = 6;
+
+  // tradable_supplies is the list of credit batch tradable supply.
+  repeated Supply tradable_supplies = 7;
+
+  // retired_supplies is the list of credit batch retired supply.
+  repeated Supply retired_supplies = 8;
```

Comment:
```suggestion
  repeated Balance balances = 5;

  // tradable_supplies is the list of credit batch tradable supply.
  repeated Supply supplies = 6;

```

Reply:
Would this mean adding an enum to the `Balance` type to keep track of if it's retired or tradable?

Reply:
updated

Reply:
I just spoke with @aaronc about this and actually the intention of his comment was to still have separate fields for `retired_balance` and `tradable_balance`, but not have them be entirely separate `Balance` / `Supply` messages (for tradable / retired).

I think enums over complicate things, and we don't want/need to be tracking any balances outside of those that are retired or tradable (e.g. cancelled credits should never be tracked in genesis import/export).

So this would look something more like:
```proto
message Balance {
   // address is the account address of the account holding credits.
   string address = 1;

   // batch_denom is the unique ID of the credit batch.
   string batch_denom = 2;

   // tradable_balance is the tradable balance of the credit batch.
   string tradable_balance = 3;

   // retired_balance is the retired balance of the credit batch.
   string retired_balance = 3;
 }
 ```

Reply:
retired balance ID  should be `4`,  right?